src/components/GridCell/GridCell.js

```diff
@@ -1,32 +1,42 @@
-import React , { Component } from 'react';
+import React, { Component } from 'react';
 
-class GridCell extends Component { 
-    constructor(props){
+class GridCell extends Component {
+    constructor(props) {
         super(props);
 
         this.state = {
-            fill: '#ffffff'
+            fill: '#ffffff',
+            selected: false
         }
 
+        this.select = this.select.bind(this);
         this.paintCell = this.paintCell.bind(this);
         this.handleDown = this.handleDown.bind(this);
         this.handleOver = this.handleOver.bind(this);
         this.handleUp = this.handleUp.bind(this);
     }
 
+    select() {
+        if (this.props.mode === 'select') {
+            this.setState({
+                selected: true
+            })
+        }
+    }
+
```

Comment:
this function lets you select a cell by changing its state to selected - the state influences the css and the appearance of being selected